manifests/init.pp

```diff
@@ -376,12 +376,27 @@
     if $agent_major_version != undef and $agent_major_version != $_agent_major_version {
       fail('Provided and deduced agent_major_version don\'t match')
     }
+    $_agent_minor_version = 0 + $3
   } elsif $agent_major_version != undef {
     $_agent_major_version = $agent_major_version
   } else {
     $_agent_major_version = $datadog_agent::params::default_agent_major_version
   }
 
+  case $facts['os']['name'] {
+    'RedHat', 'CentOS', 'OracleLinux': {
```

Comment:
Wondering whether this logic would live better in [manifests/redhat.pp](https://github.com/DataDog/puppet-datadog-agent/blob/pythyu/centos_and_gpg_puppet_update/manifests/redhat.pp) instead?